Lib/unittest/main.py

```diff
@@ -39,7 +39,7 @@ def _convert_name(name):
             name = rel_path
         # on Windows both '\' and '/' are used as path
         # separators. Better to replace both than rely on os.path.sep
-        return name[:-3].replace('\\', '.').replace('/', '.')
+        return name[:-3].replace('\\', '.').replace('/', '.').lstrip('.')
```

Comment:
I'd prefer we do `name.removeprefix("./")` (and the equivalent for Windows paths) *before* we replace away all the slashes. This is a path operation, so we should do it while the string still represents a path, not a module name.

Reply:
So before committing and testing, I'm presuming this will look like:

```py
    if os.path.isfile(name) and name.lower().endswith('.py'):
        if os.path.isabs(name):
            rel_path = os.path.relpath(name, os.getcwd())
            if os.path.isabs(rel_path) or rel_path.startswith(os.pardir):
                return name
            name = rel_path.removeprefix('./').removeprefix('.\\')
        # on Windows both '\' and '/' are used as path
        # separators. Better to replace both than rely on os.path.sep
        return name[:-3].replace('\\', '.').replace('/', '.')
    return name
```
I think you could be right and this might have broken something, despite passing tests.

Let me know if this works.

Reply:
Wait a second, that's still wrong. I'll rewrite!

Reply:
I don't think it should be only in the isabs() branch, absolute paths don't start with `./`.

Reply:
```py
        return name[:-3].removeprefix('./').removeprefix('.\\').replace('\\', '.').replace('/', '.')
```
That's more like it. A bit long.